docs/apm/troubleshooting.asciidoc

```diff
@@ -58,6 +63,66 @@ Navigate to *APM* > *Settings* > *Indices*, and change all `apm_oss.*Pattern` va
 include the new index pattern. For example: `customIndexName-*`.
 
 [float]
+[[troubleshooting-too-many-transactions]]
+=== Too many unique transaction names
+
+Transaction names are defined in each APM Agent; when an Agent supports a framework,
+it includes logic for naming the transactions that the framework creates.
+In some cases though, like when using an Agent's API to create custom transactions,
+it is up to the user to define a pattern for transaction naming.
+When transactions are named incorrectly, each unique URL can be associated with a unique transaction group—causing
+an explosion in the number of transaction groups per service, and leading to inaccuracies in the APM app.
+
+To fix a large number of unique transaction names,
+you need to change how you are using the Agent API to name your transactions.
+To do this, ensure you are **not** naming based on parameters that can change.
+For example, user ids, product ids, order numbers, query parameters, etc.,
+should be stripped away, and commonality should be found between your unique URLs.
+
+Let's look at an example from the RUM Agent documentation. Here are a few URLs you might find on Elastic.co:
+
+[source,yml]
+----
+// Blog Posts
+https://www.elastic.co/blog/reflections-on-three-years-in-the-elastic-public-sector
+https://www.elastic.co/blog/say-heya-to-the-elastic-search-awards
+https://www.elastic.co/blog/and-the-winner-of-the-elasticon-2018-training-subscription-drawing-is
+
+// Documentation
+https://www.elastic.co/guide/en/elastic-stack/current/index.html
+https://www.elastic.co/guide/en/apm/get-started/current/index.html
+https://www.elastic.co/guide/en/infrastructure/guide/current/index.html
+----
+
+These URLs, like most, include unique names.
+If we named transactions based on each unique URL, we'd end up with the problem described above—a
+very large number of different transaction names.
+Instead, we should strip away the unique information and group our transactions based on common information.
+In this case, that means naming all blog transactions, `/blog`, and all documentation transactions, `/guide`.
```

Comment:
This example is also in the RUM documentation. It seemed useful to explain it here as well.